plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java

```diff
@@ -534,7 +534,7 @@ private void setPreviousSelection() {
         targetPath = targetPath.substring( 7 );
       }
       java.io.File filePath = new java.io.File( targetPath );
-      if ( !filePath.exists() && !ProviderFilterType.REPOSITORY.toString().equalsIgnoreCase( fileDialogOperation.getProvider() ) ) {
+      if ( !filePath.exists() && ProviderFilterType.LOCAL.toString().equalsIgnoreCase( fileDialogOperation.getProvider() ) ) {
```

Comment:
Why was the ProviderFilterType set to Repository

Reply:
it used to say: "If the provider is anything other than REPOSITORY, and the (java.io.) File doesn't exist, return"

That broke because not the only non-REPOSITORY provider that actually works with java.io.File is LOCAL. Other providers, like VFS (this bug) would just never return true for "exists()". 

So the new check is: "if the provider is LOCAL and the File doesn't exist, return"